dev/dev_setup.sh

```diff
@@ -0,0 +1,321 @@
+#!/bin/sh
+
+# cryptol setup script
+#
+# Supported distribution(s):
+#  * macOS 14 (AArch64)
+#  * Ubuntu 20.04 (x86_64)
+#  * Ubuntu 22.04 (x86_64)
+#
+# This script installs everything needed to get a working development
+# environment for cryptol. Any new environment requirements should be
+# added to this script. This script is not interactive but it may define
+# some environment variables that need to be loaded after it runs.
+# For a list of the setup operations performed by this script, see the
+# "Main Operations" section near the end of this file.
+# Assumptions:
+# - In Ubuntu, we assume that the system has already run `apt-get update`
+# - In macOS, we assume the system has `brew` installed and on the path
+#
+# There is some half-baked support for macOS 12 (x86_64),
+# but it hasn't been tested.
+#
+
+set -e
+
+HERE=$(cd `dirname $0`; pwd)
+LOG=$HERE/dev_setup.log
+truncate -s 0 $LOG
+ROOT=$HERE
+
+# Create a new, empty file for environment variables
+VAR_FILE=$ROOT/env.sh
+truncate -s 0 $VAR_FILE
+echo "# Generated environment variables. Manual changes will get clobbered by dev_setup.sh" >> $VAR_FILE
+
+GHCUP_URL="https://downloads.haskell.org/~ghcup"
+GHC_VERSION="9.4.8"
+CABAL_VERSION="3.10.3.0"
+
+WHAT4_SOLVERS_SNAPSHOT="snapshot-20240212"
+WHAT4_SOLVERS_URL="https://github.com/GaloisInc/what4-solvers/releases/download/$WHAT4_SOLVERS_SNAPSHOT"
+WHAT4_SOLVERS_MACOS_12="macos-12-X64-bin.zip"
+WHAT4_SOLVERS_MACOS_14="macos-14-ARM64-bin.zip"
+WHAT4_SOLVERS_UBUNTU_20="ubuntu-20.04-X64-bin.zip"
+WHAT4_SOLVERS_UBUNTU_22="ubuntu-22.04-X64-bin.zip"
+WHAT4_CVC4_VERSION="version 1.8"
+WHAT4_CVC5_VERSION="version 1.1.1"
+WHAT4_Z3_VERSION="version 4.8.14"
+
+# Set of supported platforms:
+MACOS14="macos14"
+MACOS12="macos12" # actually, this isn't supported yet
+UBUNTU20="ubuntu-20.04"
+UBUNTU22="ubuntu-22.04"
+
+USED_BREW=false
+
+# Returns a string indicating the platform (from the set above), or empty if
+# a supported platform is not detected.
```

Comment:
I would suggest returning a non-blank value here (e.g. maybe "unknown-platform" or something like that).  The reason is that blank results can trip up shell scripts unless special care is taken to guard against their presence (e.g. `case $blank_value in ...`  becomes `case in ...` and `if [ ubuntu -eq $BLANK ] ...` becomes `if [ ubuntu -eq ] ...`).  I don't know that to be the case herein, it's just a preventative measure.

Reply:
When we call this function we check right away if it's blank and `exit` if so, so it shouldn't cause issues anywhere else.